shared/src/jsMain/kotlin/com/github/jetbrains/rssreader/app/RssReaderJSViewModel.kt

```diff
@@ -0,0 +1,87 @@
+package com.github.jetbrains.rssreader.app
+
+import com.github.jetbrains.rssreader.core.entity.Feed
+import com.github.jetbrains.rssreader.core.entity.Post
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.get
+
+@ExperimentalJsExport
+@JsExport
+class FeedStateJsWrapper(
+    val progress: Boolean,
+    val feeds: Array<FeedJsWrapper>,
+    val selectedFeed: Feed? = null
+)
+
+@ExperimentalJsExport
+@JsExport
+class FeedJsWrapper(
+    val title: String,
+    val link: String,
+    val desc: String,
+    val imageUrl: String?,
+    val posts: Array<Post>,
+    val sourceUrl: String,
+    val isDefault: Boolean
+)
+
+fun FeedState.toJsObject() = FeedStateJsWrapper(
+    progress,
+    feeds.map {
+        Feed(
+            it.title,
+            it.link,
+            it.desc,
+            it.imageUrl,
+            it.posts,
+            it.sourceUrl,
+            it.isDefault
+        ).toJsObject()
```

Comment:
it.toJsObject() ?

Reply:
I was under the impression that I also needed to wrap the `Feed` itself to export the array of `Post` inside it. 
```kotlin
val posts: Array<Post>
```

Which works nicely on the JS side because now we can access the `Post` objects in native JS arrays directly
```JavaScript
setPosts(state.feeds[0]?.posts)
```

Am I missing something ? (sorry)

Reply:
Haha yes, got it... still a Kotlin learner here :)

Reply:
🤦 of course! thank you